cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutor.java

```diff
@@ -343,6 +379,9 @@ private void safeShutdown() {
                 // ignore this
             }
             process = null;
+
+            // send health check that process is no longer alive
+            healthCheck(driver);
```

Comment:
Generally yes - but that way the scheduler immediately knows that the server is dead.
But you're right - I forgot some code to actually update the current health check when it receives that the task is no longer running.